Be able to specify a static key for some set of files (essential if using HTML5 appcache manifest)

cbxp opened the pull request:
E.g. #{press.compressed-script 'pressed-artifact.js'} and #{press.compressed-stylesheet 'pressed-artifact.css'}

Commit:
Added support for human readable fixed name of compressed resources - optional argument in #{press.compressed-script 'pressed-artifact.js'} and #{press.compressed-stylesheet 'pressed-artifact.css'}. Useful when one want to add compressed resources for caching by browser, for example in .appcache

Commit:
add media=screen to all compressed tags (until we get a better solution either from upstream or implement ourselves) - this is needed to differentiate from media=print stylesheets

Commit:
make sure press works with the play pdf module

iammerrick commented:
This is essential if you want to be multi-tenant. Press currently locks you into a single server, this would mitigate that problem.

dirkmc commented:
Hi merrick, sorry for late response, I was on vacation. The play guys are no longer taking updates to plugins for play version 1, so I'm no longer maintaining this plugin. Thanks for checking this out anyway.
Dirk

Commit:
Merge branch 'master' of https://github.com/dirkmc/press

Commit:
use play's VirtualFiles to resolve less imports from modules

Commit:
more effective list clone

Commit:
version bumped to 1.1

Commit:
adding build script for jenkins

Commit:
resolve deps from maven instead of keeping them in lib
keep less at 1.3.3 because yuicompressor doesn't work with newer rhino that it requires

Commit:
move sources to src so that play will not recompile these classes all the time

angryziber committed:
use newer yuicompressor 2.4.8 that is not available in maven

angryziber committed:
bugfix: make sure less imports are actually cached

angryziber committed:
strip windows CR chars - they seem to break less.js

angryziber committed:
releasing 1.2.2 with CR newline fix

angryziber committed:
releases play 1.3 that can use lessc on the command-line instead of the built-in Rhino based mega slow less

angryziber committed:
Play.getVirtualFile() returns null in case file is not found, so we must handle this correctly

angryziber committed:
more logical access modifiers

angryziber committed:
log less errors properly

angryziber committed:
reuse press' own compressed file storage for less files (e.g. don't store compiled less files in cache if disk storage is enabled)

angryziber committed:
use virtual files for resolving of import last modified times - otherwise we miss inherited files

angryziber committed:
do not use virtual file ro resolve tmp location for compressed files, so that absolute paths are possible, and put it under tmp by default

angryziber committed:
use shorter name for compiled less files

angryziber committed:
release play-press 1.3.1

angryziber committed:
release 1.3.2 without outputting the original filenames and then scanning the output
use LinkedHashMap instead

angryziber committed:
enable conf watching for preprod as well

angryziber committed:
support Windows better with cross-platform separator

angryziber committed:
bump version

Commit:
release play-press 1.3.4 with media attribute support

andrei-codeborne committed:
Fixed bug: add "media" attribute also in non-compressing mode; added tests.

andrei-codeborne committed:
Added test for Plugin.addCss()

andrei-codeborne committed:
#{press.single-stylesheet} also supports media attribute; added tests.

andrei-codeborne committed:
use relative path in unit-tests to make them runnable within other projects

andrei-codeborne committed:
use relative path in unit-tests to make them runnable within other projects

andrei-codeborne committed:
fixed unit-test: it should not depend on "Play.mode" that could be set by another test

asolntsev committed:
Fixed IDEA warnings

asolntsev committed:
Added test for ScriptCompressor and StyleCompressor

asolntsev committed:
Release play-press 2.0 that uses Google Closure instead of YUI Compressor for minifying JavaScript files

asolntsev committed:
fixed play-press tests to avoid dependency on test execution order

andrei-codeborne committed:
updated module exclusions

asolntsev committed:
ignore /dist and /tmp folders

asolntsev committed:
Merge remote-tracking branch 'origin/master'

Conflicts:
	.gitignore

asolntsev committed:
ignore /dist and /tmp folders

Commit:
Log Play roots if less file is not found

Commit:
Added test for LESS compilation with Rhino Javascript engine

Commit:
upgrade play-press version to 2.1: better logging of missing files

angryziber committed:
add support for precompiled less files on production

angryziber committed:
release play 2.2 with support for precompiles less files in prod mode

angryziber committed:
make sure PlayLessEngine is not created on every request

angryziber committed:
create PleyLessEngine once, not twice

angryziber committed:
check for precompiled files in dev as well

angryziber committed:
release 2.2.1

angryziber committed:
add logs

angryziber committed:
disable debug logs

asolntsev committed:
play-press 2.2.4: do not reset play properties

angryziber committed:
mark press controller with @NoTransaction, so that it will not fail when DB is not available

Commit:
Properly close resources in finally block

angryziber committed:
get rid of built-in old rhino in yui-compressor (next step - git rid of yui compressor)

angryziber committed:
Merge branch 'master' of github.com:codeborne/play-press

angryziber committed:
release play press 2.3 with stripped yui-compressor (not including outdated rhino)

asolntsev committed:
play-press 2.3.1: remove mockito dependency (it's part of play framework 1.3.x)

asolntsev committed:
play-press 2.3.1: package jars/yuicompressor-2.4.8.jar (not declared as dependency, but used in shell-scripts)

Commit:
play-press 2.2.6: remove mockito jar from zip (as it's test dependency)

asolntsev committed:
Merge remote-tracking branch 'origin/master'

Conflicts:
	conf/dependencies.yml

Commit:
play-press 2.2.7: bugfix: it's important close OnDiskCompressedFile because it renames tmp file

asolntsev committed:
Merge remote-tracking branch 'origin/master'

angryziber committed:
support for single-quotes in less @import statements

Commit:
play-press 2.2.9: correct usage of slf4j logger

Commit:
fixing reloading of child less files - resolve relative virtual paths properly when building a lists of imports

Commit:
speed up less import resolving

Commit:
bump version to 2.3 - better reloading of nested less files

Commit:
use version 2.3.1

Commit:
bump major version to avoid conflicts with old experimental 2.3 branch

angryziber committed:
drop unused file

angryziber committed:
make node less engine more configurable

angryziber committed:
play press 2.5 with more configurable node less runner

angryziber committed:
remove dependencies on yuicompressor and clojure compiler - we compress for production during the build

angryziber committed:
drop bundled outdated less 1.3.3, always default to command-line lessc; bump version to 3.0

angryziber committed:
play tries to resolve tags with .html extension first - renaming so they are a bit faster

angryziber committed:
do not fail in case of absence of lessc in precompiled mode

angryziber committed:
no need for old newline workaround

angryziber committed:
make sure lessc errors are properly reported

angryziber committed:
bump version to 3.1

asolntsev committed:
Merge branch 'master' of github.com:codeborne/play-press

andrei-codeborne committed:
play-press 3.2: do not clear CSS/JS cache on application restart

production users must be able to use cached resources for a long time

andrei-codeborne committed:
run tests with every build

asolntsev committed:
fix failing unit-tests in play-press module

asolntsev committed:
fix failing unit-tests in play-press module

asolntsev committed:
do not delete test reports in the end of build

asolntsev committed:
play-press 3.3: log key in case of error

asolntsev committed:
play-press 3.3: code cleanup

asolntsev committed:
play-press 3.3: report file name and exit code in case of lessc error

asolntsev committed:
play-press 3.3: do not return "debug" CSS file with error description to avoid caching it in live

asolntsev committed:
ignore tmp files
